app/electron/main.ts

```diff
@@ -686,23 +700,55 @@ function startElecron() {
         return;
       }
 
-      const child: ChildProcessWithoutNullStreams = spawn(
-        eventData.command,
-        eventData.args,
-        eventData.options
-      );
+      if (!eventData.options) {
+        eventData.options = {};
+      }
 
-      child.stdout.on('data', (data: string | Buffer) => {
-        event.sender.send('command-stdout', eventData.id, data.toString());
-      });
+      // if command is electron-node use fork
+      if (eventData.command === 'electron-node') {
+        if (!eventData.options?.env) {
+          eventData.options.env = {};
+        }
+        eventData.options.env['ELECTRON_RUN_AS_NODE'] = '1';
+        eventData.options.silent = true;
+        const child: ChildProcess = fork(
+          eventData.args[0],
+          eventData.args.slice(1),
+          eventData.options
+        );
 
-      child.stderr.on('data', (data: string | Buffer) => {
-        event.sender.send('command-stderr', eventData.id, data.toString());
-      });
+        if (child.stdout !== null) {
+          child.stdout?.on('data', (data: string | Buffer) => {
+            event.sender.send('command-stdout', eventData.id, data.toString());
+          });
+        }
+        if (child.stderr !== null) {
+          child.stderr.on('data', (data: string | Buffer) => {
+            event.sender.send('command-stderr', eventData.id, data.toString());
+          });
+        }
+        child.on('exit', (code: number | null) => {
+          event.sender.send('command-exit', eventData.id, code);
+        });
+      } else {
```

Comment:
I understand why you are special casing this, but I think we can things more clear and future proof if we have an IPC call just for plugins, and when we wrap it in a nicer `Headlamp.` API as suggested in https://github.com/headlamp-k8s/plugins/pull/35#discussion_r1567568666